aptos-move/e2e-move-tests/src/tests/aggregator_v2.rs

```diff
@@ -100,105 +104,214 @@ fn test_aggregators_e2e() {
+                (1usize..len)
+                    .prop_flat_map(move |first| {
+                        // Second is non-empty, but can finish the block: [1, len - first]
+                        (Just(first), 1usize..len - first + 1)
+                    })
+                    .boxed()
+            }
+        })
+        .boxed()
+}
 
-    let txn1 = new(&mut h, &acc, 0, 600);
-    let txn2 = try_add(&mut h, &acc, 0, 400);
-    let txn3 = try_sub(&mut h, &acc, 0, 500);
+proptest! {
+    #![proptest_config(ProptestConfig {
+        // Cases are expensive, few cases is enough.
+        cases: 5,
+        result_cache: prop::test_runner::basic_result_cache,
+        .. ProptestConfig::default()
+    })]
 
-    assert_success!(h.run(txn1));
-    assert_success!(h.run(txn2));
+    #[test]
+    fn test_aggregator_lifetime((first_block, second_block) in arb_block_split(13)) {
+        println!("Testing test_aggregator_lifetime {} {}", first_block, second_block);
 
-    // Value dropped below zero - abort with EAGGREGATOR_UNDERFLOW.
-    assert_abort!(h.run(txn3), 131074);
-}
+        let (mut h, acc) = setup(false);
 
-#[test]
-fn test_aggregator_materialize_underflow() {
-    let (mut h, acc) = setup();
+        let txns = vec![
+            (0, new(&mut h, &acc, 0, 1500)),
+            (0, add(&mut h, &acc, 0, 400)),
+            (0, materialize(&mut h, &acc, 0)),
+            (0, add(&mut h, &acc, 0, 500)),
+            (0, check(&mut h, &acc, 0, 900)),
+            (0, materialize_and_add(&mut h, &acc, 0, 600)),
+            (0, materialize_and_sub(&mut h, &acc, 0, 600)),
+            (0, check(&mut h, &acc, 0, 900)),
+            (0, sub_add(&mut h, &acc, 0, 200, 300)),
+            (0, check(&mut h, &acc, 0, 1000)),
+            // These 2 transactions fail, and should have no side-effects.
+            (131073, add_and_materialize(&mut h, &acc, 0, 501)),
+            (131073, sub_and_materialize(&mut h, &acc, 0, 1001)),
+            (0, check(&mut h, &acc, 0, 1000)),
+        ];
+        run_block_in_parts(
+            &mut h,
+            first_block,
+            second_block,
+            txns,
+        );
+    }
 
-    let txn1 = new(&mut h, &acc, 0, 600);
-    let txn2 = materialize_and_try_sub(&mut h, &acc, 0, 400);
+    #[test]
+    fn test_aggregator_underflow((first_block, second_block) in arb_block_split(3)) {
+        println!("Testing test_aggregator_underflow {} {}", first_block, second_block);
 
-    // Underflow on materialized value leads to abort with EAGGREGATOR_UNDERFLOW.
-    assert_success!(h.run(txn1));
-    assert_abort!(h.run(txn2), 131074);
-}
+        let (mut h, acc) = setup(false);
 
-#[test]
-#[should_panic]
-fn test_aggregator_overflow() {
-    let (mut h, acc) = setup();
+        let txns = vec![
+            (0, new(&mut h, &acc, 0, 600)),
+            (0, add(&mut h, &acc, 0, 400)),
+            // Value dropped below zero - abort with EAGGREGATOR_UNDERFLOW.
+            (131074, sub(&mut h, &acc, 0, 500))
```

Comment:
Should we call `try_add`, `try_sub` instead? 
Also, `try_sub` would return false here instead of an error code right?

Reply:
tests you copied are from v1 version. So having them be identical - is starting way to get them to run. It doesn't make much difference for these tests.

will add separately some new tests, where we can use try_add/try_sub